hax/treemakers/CorrectedDoubleS1Scatter.py

```diff
@@ -217,131 +176,153 @@ def extract_data(self, event):
         result['ds_s1_dt'] = peaks[s1_b].center_time - peaks[s1_a].center_time
-
         # Need the observed ('uncorrected') position.
         # pax Interaction positions are corrected so lookup the
         # uncorrected positions in the ReconstructedPosition objects
         for rp in peaks[s2_a].reconstructed_positions:
             if rp.algorithm == 'PosRecNeuralNet':
-                result['int_a_x_nn'] = rp.x
-                result['int_a_y_nn'] = rp.y
-                result['int_a_r_nn'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
-                int_a_x_observed = rp.x
-                int_a_y_observed = rp.y
+                result['int_a_x_observed_nn'] = rp.x
+                result['int_a_y_observed_nn'] = rp.y
+                result['int_a_r_observed_nn'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
+            elif rp.algorithm == 'PosRecTopPatternFit':
+                result['int_a_x_observed_tpf'] = rp.x
+                result['int_a_y_observed_tpf'] = rp.y
+                result['int_a_r_observed_tpf'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
         for rp in peaks[s2_b].reconstructed_positions:
             if rp.algorithm == 'PosRecNeuralNet':
-                result['int_b_x_nn'] = rp.x
-                result['int_b_y_nn'] = rp.y
-                result['int_b_r_nn'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
-
+                result['int_b_x_observed_nn'] = rp.x
+                result['int_b_y_observed_nn'] = rp.y
+                result['int_b_r_observed_nn'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
+            elif rp.algorithm == 'PosRecTopPatternFit':
+                result['int_b_x_observed_tpf'] = rp.x
+                result['int_b_y_observed_tpf'] = rp.y
+                result['int_b_r_observed_tpf'] = np.sqrt(rp.x ** 2 + rp.y ** 2)
         int_a_z = interactions[int_a].z - interactions[int_a].z_correction
         result['int_a_z_observed'] = int_a_z
         int_b_z = interactions[int_b].z - interactions[int_b].z_correction
         result['int_b_z_observed'] = int_b_z
 
-        # Correct S2_a. No correction for S2_b because, S2_b is mostly backgroud events, or S2_b ==S2_a
-        cvals = [int_a_x_observed, int_a_y_observed]
-        result['s2_a_xy_correction_tot'] = (1.0 /
-                                            self.corrections_handler.get_correction_from_map(
-                                                "s2_xy_map", self.run_number, cvals))
-        result['s2_a_xy_correction_top'] = (1.0 /
-                                            self.corrections_handler.get_correction_from_map(
-                                                "s2_xy_map", self.run_number, cvals, map_name='map_top'))
-        result['s2_a_xy_correction_bottom'] = (1.0 /
-                                               self.corrections_handler.get_correction_from_map(
-                                                   "s2_xy_map", self.run_number, cvals, map_name='map_bottom'))
-
+        int_signal = ['int_a_', 'int_b_']
```

Comment:
ideally you can use this for loop for above nn&tpf positions, but it's fine as is.